### PR TITLE
replace webpack with Rspack for the main app

### LINKED ISSUE #47025: migrate to rspack

### PR DESCRIPTION
**Pull request by uladzimirdev**
Closes https://github.com/metabase/metabase/issues/47025

### Description

[Rspack](https://rspack.dev/) is faster than webpack, it's written in Rust as other modern tools like [swc](https://swc.rs/).  

Why Rspack.

- it's 3 times faster than webpack on our app
- it supports same plugins as webpack or provides replacements
- it support the same config, so we mostly copy old webpack config
- it's [v1.0](https://rspack.dev/blog/announcing-1-0), kinda marked as production ready

### How to verify

Before approval please run it locally, try to do the same task you do every day and report feedback if any

### Demo

Rspack

image

Webpack

image

<details><summary>Detailed demo</summary>
<p>

### Before

https://github.com/user-attachments/assets/e0b3df7c-b340-421a-9c9b-5e1ed61ed537

### After

https://github.com/user-attachments/assets/035bba10-c41b-421e-978c-d8c398c60602

### cljs live reload

https://github.com/user-attachments/assets/120f2375-65d7-42bd-a86a-f66464af9981

</p>
</details> 

### Checklist

- [x] refresh (hot) mode works
- [x] prod build works
- [x] ci is green

### CODE REVIEW

File: frontend/src/metabase/lib/dom.js [hunk 1]
```diff
@@ -495,23 +495,10 @@ export function initializeIframeResizer(onReady = () => {}) {
...
-      return false;
-    }
-
     // Make iframe-resizer avaliable to the embed
     // We only care about contentWindow so require that minified file
 
-    require.ensure([], require => {
-      require("iframe-resizer/js/iframeResizer.contentWindow.js");
-    });
+    import("iframe-resizer/js/iframeResizer.contentWindow.js");
```
Comment (WiNloSt) [inline frontend/src/metabase/lib/dom.js:501]: I couldn't test this line due to [iFrameResizer doesn't work on master (v51)](https://github.com/metabase/metabase/issues/47061#top).

**Review by WiNloSt — COMMENTED** [review]
I haven't fully tested yet, but overall, the build is pretty fast. I'm waiting to fix the iframe resizer bug before testing this PR. If you need to merge it before then, then it's fine FYI.

**Comment by iethree** [comment]
works for me in local dev!

**Comment by romeovs** [comment]
Works for me too locally!

File: frontend/src/metabase/lib/dom.js (hunk 1 above)
Comment (WiNloSt) [reply frontend/src/metabase/lib/dom.js:501]: @uladzimirdev I haven't done with the fix yet but this particular line fixes the issue where `require.ensure` doesn't actually import `iframe-resizer/js/iframeResizer.contentWindow.js` which is a part of the problem of the mentioned issue.

That being said, I approve this line from embedding's point of view.

Comment (uladzimirdev) [reply frontend/src/metabase/lib/dom.js]: fixed separately

**Comment by rafpaf** [comment]
This is exciting! I checked out this branch and I'm seeing a configuration error mentioned in the logs. Is it related?

```
[js] husky - Git hooks installed
[js] Configuration error:
[js] - Expected boolean, received object at "cache"
[js] <i> [webpack-dev-server] Project is running at:
[js] <i> [webpack-dev-server] Loopback: http://localhost:8080/
[js] <i> [webpack-dev-server] On Your Network (IPv4): http://192.168.0.251:8080/
[js] <i> [webpack-dev-server] On Your Network (IPv6): http://[fe80::1]:8080/
[js] <i> [webpack-dev-server] Content not from webpack is served from '/Users/raf/metabase/public' directory
[js] Rspack compiled successfully
```

**Comment by uladzimirdev** [comment]
@rafpaf thanks for the feedback!

> Rspack currently only supports memory cache, persistent cache is under planning, for more details refer to [Persistent Cache Support](https://rspack.dev/misc/planning/roadmap#persistent-cache-support).

I dropped object configuration, so you can stop using `FS_CACHE` env variable

**Review by kamilmielnik — APPROVED** [review]
:1st_place_medal:

**Pull request merged**